app/packages/components/src/components/PillBadge/PillBadge.tsx

```diff
@@ -10,12 +10,16 @@ const PillBadge = ({
   variant = "filled",
   showIcon = true,
   operator,
+  readOnly = false,
```

Comment:
Why is this needed for PillBadge?

To set a readOnly PillBadge view you simply pass down a singular string. If only a string is given to this component (with or without a color) it becomes unselectable and is no longer a writeable component

Reply:
if we simply use 
```
schema = {"text": "Disabled", "color": "secondary"}
badge = types.PillBadgeView(**schema)
```

The badge doesn't reflect it is a dropdown select - the downward caret will be missing. disabled / readOnly are legit properties to use in common components. if you really don't like the readOnly prop, happy to huddle and resolve!

Reply:
But isn't that the point of readOnly? If I should only be reading the state of something, I don't understand why they would need to see the dropdown caret?

Having a dropdown caret implies that they can change the state of the pill, but we don't want that

So we should just show the pill and whatever the current status is without the ability to change it - which is just the string & a color choice